### PR TITLE
Automate release notes

### PR DESCRIPTION
**Pull request by Jonas-Isr**
## Context

AI/ai-sdk-java-backlog#94

We need release notes now that we have an initial version publicly available. For this, I copied the way Cloud SDK is doing their release notes:

- There is now a `release-notes.md` file in under `docs/release-notes`. In a PR, you can add something to this file. This file functions as a collection of all release notes since the last (minor or major) release.
- (Once a new minor or major version is published) There is a file named `release-notes-0-to-14.mdx` under `docs/release-notes` that captures all the release notes since the last major release. After every 15 minor releases, a new file is used to not avoid making this file too long (the next files will then be called `release-notes-15-to-29.mdx` and so on).
- In the `Prepare Release` workflow, content from `release-notes.md` is copied to the appropriate `release-notes-X-to-Y.mdx` file. After the release version is created, a new (empty) `release-notes.md` is created for the next version.

### Feature scope:
 
- [x]  GitHub release contains release notes (GA itself can be a one time manual job)
    - To make this work for the GA, we could add release note into `release-notes.md`.
- [x] Process is automated (At least the step of collecting the release notes for the current release should be automated)
- [x] Concept for how diff between latest release and current status can be viewed 
    - This can be seen in the `release-notes.md` file.

## Definition of Done

- [x] Functionality scope stated & covered
- [ ] ~Tests cover the scope above~
- [ ] ~Error handling created / updated & covered by the tests above~
- [ ] ~Aligned changes with the JavaScript SDK~
- [ ] ~Documentation updated~
- [ ] ~Release notes updated~

### CODE REVIEW

File: .pipeline/scripts/release_notes_template.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## 0.X.0-SNAPSHOT
```
Comment (newtork) [inline .pipeline/scripts/release_notes_template.md:1]:
```suggestion
## 1.X.0
```

Wouldn't this make more sense?

File: .github/workflows/prepare-release.yaml [hunk 2]
```diff
@@ -148,7 +158,7 @@ jobs:
           
           PR_URL=$(gh pr create --title "Release ${{ needs.bump-version.outputs.release-version }}" --body "## TODOs
           - [ ] Review the changes in [the release commit]($COMMIT_URL)
-          - [ ] Update the Release notes
+          - [ ] If major release: Update header of .pipeline/scripts/release_notes_template.md
```
Comment (newtork) [inline .github/workflows/prepare-release.yaml:161]: I'm not sure we'll need a dedicated reminder here that comes up with every minor release🤔

**Comment by Jonas-Isr** [comment]
Successful run in my fork: https://github.com/Jonas-Isr/ai-sdk-java-test/pull/16

File: .pipeline/scripts/release_notes_automation.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    date = datetime.today().strftime('%B %d, %Y')
+    # Replace the first line with: ## 5.2.0 - January 17, 2024
+    file = re.sub("^## .*", "## " + version + " - " + date, file)
+    return file
+
+def link_github_release(file, version):
+    old_github_release_link = "\[All Release Changes\]\(https://github.com/SAP/ai-sdk-java/releases/\)"
+    new_github_release_link = "[All Release Changes](https://github.com/SAP/cloud-sdk-java/releases/tag/rel%2F"+version+")"
+    file = re.sub(old_github_release_link, new_github_release_link, file)
+    return file
```
Comment (newtork) [inline .pipeline/scripts/release_notes_automation.py:58]: (Question/Major)
This looks strange to me, why would there be a link to SAP Cloud SDK?

File: .pipeline/scripts/release_notes_automation.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        for line in file:
+            if releases_pattern.match(line):
+                count += 1
+    return count
+
+def find_target_file(version):
+    # release-notes-X-to-Y.mdx with every 15 versions the index increases by 15 and stays the same for 15 versions
+    minor_version = int(version.split(".")[1])
+    index = minor_version // 15 * 15
+    return "release-notes-" + str(index) + "-to-" + str(index + 14) + ".mdx"
```
Comment (newtork) [inline .pipeline/scripts/release_notes_automation.py:75]: (Question)

I don't understand this logic, isn't `index == minor_version` always?

File: .pipeline/scripts/release_notes_automation.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+""",
+"""### 📈 Improvements
+
+-
+
+""",
+"""### 🐛 Fixed Issues
+
+- 
+"""]
```
[the remark this answers is not in the document]
Comment (CharlesDuboisSAP) [reply .pipeline/scripts/release_notes_automation.py:40]: It has never needed maintenance in the Cloud SDK, I think it's fine

File: .pipeline/scripts/release_notes_automation.py (hunk 3 above)
Comment (Jonas-Isr) [reply .pipeline/scripts/release_notes_automation.py:58]: You are correct, I missed to adapt this link when copying from Could SDK.

File: .pipeline/scripts/release_notes_automation.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Jonas-Isr) [reply .pipeline/scripts/release_notes_automation.py:40]: I am not experienced with regexes. Would that really be less maintenance effort than having an explicit list?

[the remark this answers is not in the document]
Comment (Jonas-Isr) [reply .pipeline/scripts/release_notes_automation.py:40]: Also note, that you can always add your own custom categories in `release_notes.md` and they will be copied. This here only ensures that no empty category (of the 4 suggested ones in the template) is copied over into the `release_notes_X_to_Y` files.

File: .github/workflows/prepare-release.yaml (hunk 2 above)
Comment (Jonas-Isr) [reply .github/workflows/prepare-release.yaml]: Deleted the reminder.

**Review by newtork — APPROVED** [review]

**Pull request merged**